### PR TITLE
(PE-30656) Allow project_inventory_targets to use PuppetConnectData

### PR DESCRIPTION
**Pull request by donoghuc**
This commit updates the `project_inventory_targets` endpoint to accept data from puppet connect and uses it to enable the connect plugin.

### CODE REVIEW

File: lib/bolt_server/schemas/connect-data.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "title": "project_inventory_targets connect plugin data",
+  "description": "POST project_inventory_targets connect plugin data",
+  "type": "object",
+  "properties": {
+    "puppet_connect_data": {
+      "type": "object",
+      "properties": {
+        "type": "object",
+        "properties": {
+          "key": {}
```
Comment (donoghuc) [inline lib/bolt_server/schemas/connect-data.json:12]: Should we enforce this as a string here?

File: developer-docs/bolt-api-servers.md [hunk 2]
```diff
@@ -380,8 +380,26 @@ This endpoint behaves identically to the /ssh/check_node_connections endpoint, b
 
 This endpoint parses a project inventory and returns a list of target hashes. Note that the only accepetable inventoryfile location is the default `inventory.yaml` at the root of the project.
 
+### Query parameters
```
Comment (barriserloth) [inline developer-docs/bolt-api-servers.md:383]: This is a nitpick, but these new sections looks like they should be subsections of the `POST /project_inventory_targets` section, so maybe these headers should be the level below the proj_inv_targets header so they don't all look like separate sections?

**Review by barriserloth — APPROVED** [review]
Besides the one small comment, this LGTM!

**Comment by beechtom** [comment]
@donoghuc I think this needs a rebase to pick up changes in CI.

**Pull request merged**